assets/js/components/ExecutionResults/ExecutionResults.jsx

```diff
@@ -158,12 +148,16 @@ function ExecutionResults({
         onFilterChange={(newPredicates) => setPredicates(newPredicates)}
       />
       <ResultsContainer
-        catalogError={false}
+        error={catalogError || executionError}
+        errorContent={[
+          catalogError ? `Failed loading catalog: ${catalogError}` : null,
+          executionError ? `Failed loading execution: ${executionError}` : null,
+        ]}
```

Comment:
Maybe this works better?

```suggestion
        errorContent={
          <>
            catalogError && `Failed loading catalog: ${catalogError}`
            executionError && `Failed loading execution: ${executionError}`
          </>
        }
```

Reply:
You could also use the `span`s here removing the need for the `Array.isArray` check.

Reply:
The suggestion does not really work better.
In order to get something workable here's how it would end up
```
errorContent={
  <>
    {catalogError && (
      <span className="block">
        Failed loading catalog: {catalogError}
      </span>
    )}
    {executionError && (
      <span className="block">
        Failed loading execution: {executionError}
      </span>
    )}
  </>
}
```
which is a bit noisy.

I'd rather keep passing the list of messages and extract the function in `NotificationBox` as you suggested.

Reply:
Your call .)